extensions/amp-form/0.1/amp-form.js

```diff
@@ -209,39 +221,70 @@ export class AmpForm {
       this.setState_(FormState_.SUBMITTING);
       const isHeadOrGet = this.method_ == 'GET' || this.method_ == 'HEAD';
       let xhrUrl = this.xhrAction_;
-      if (isHeadOrGet) {
-        xhrUrl = addParamsToUrl(this.xhrAction_, this.getFormAsObject_());
+
+      const varSubPromises = [];
+      for (let i = 0; i < varSubsFields.length; i++) {
+        const variable = varSubsFields[i].getAttribute('default-value');
+        varSubPromises.push(
+            this.urlReplacement_.expandAsync(variable).then(value => {
```

Comment:
I meant to say that we'd need a way to prevent direct calls to var-sub service with a URL to avoid circumventing the protocol assert.